### DISCUSSION TITLE
Vim - Visual bug? - stuck at indexing

### DISCUSSION THREAD

**Discussion by Telmo-Sousa** [rust-lang/rust-analyzer#16689] [Q&A]
Hey everyone,
Today I opened up a project I am working on and I noticed that rust-analyzer was stuck <br>`rust-analyzer ⠇ Indexing 1/6 (core) 16% ` <br> For moments I thought it could be a bug on macOS but nothing had changed from last night, I created a new project but still the same so I decided to check for updates but that did not change anything. <br> I decided to hop-op into my Linux machine and the same thing was happening, even after updating it... very weird, changed to my other distro and same thing was happening... but then I noticed the rust-analyzer was working, even weirder? <br> Maybe I never noticed this before but was always there...? here is a pic tho :)
![region_20240227014457](https://github.com/rust-lang/rust-analyzer/assets/68487926/084a017a-54f7-48fe-9fae-0a05e28b02d9)

**Comment by lnicola** [comment]
What version do you have?

**Reply by Telmo-Sousa** [reply]
Sorry I should have posted that yesterday but a long night made me forget that...
```
~ > rust-analyzer --version

rust-analyzer 1.76.0 (07dca48 2024-02-04)
~ > rustup --version

rustup 1.26.0 (5af9b9484 2023-04-05)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.76.0 (07dca489a 2024-02-04)`
~ > rustc --version

rustc 1.76.0 (07dca489a 2024-02-04)
~ > cargo --version

cargo 1.76.0 (c84b36747 2024-01-18)
```

**Reply by lnicola** [reply]
I suggest trying a binary from GitHub Releases, just in case. There were some recent fixes, but I think your version predates the bugs.

**Reply by Telmo-Sousa** [reply]
Thanks for the help Inicola, I will do that as soon as I have some spare time. <br> For now it does not bother me that much as it is "working".

**Comment by Telmo-Sousa** [comment]
<img width="895" alt="Screenshot 2024-02-27 at 11 14 45" src="https://github.com/rust-lang/rust-analyzer/assets/68487926/41b499f0-d08e-4234-b003-331d857ebe62">
<img width="521" alt="Screenshot 2024-02-27 at 11 18 37" src="https://github.com/rust-lang/rust-analyzer/assets/68487926/931006bd-12e3-48f7-9018-fb7c209a2034"> <br>
well i got it to work properly one time but its back at indexing and when i run the command coc.nvim is suggesting i get a warning <br> Failed to run build scripts of some packages <br> funny enough the same thing happened on both of my devices again: <br> open a project -> it does his thing -> error -> it works -> reopen the project -> stuck at indexing -> it works anyways

**Comment by Telmo-Sousa** [comment]
I wanted to give a quick update, it seems like it is a visual bug and when I enter command-line mode ":" it seems like the visual "stuck" bug updates to what rust-analyzer is doing now.
<br> Here is a video showcasing that:
[https://streamable.com/xwcof9](
https://streamable.com/xwcof9)
<br> In the video I open a Rust file with Vim, it starts indexing, cargo check runs smoothly, I close nerdtree, I enter CL mode and the rust-analyzer status updates, I do that repeatedly until it's status is only rust-analyzer. <br> It is a visual bug I guess because the same thing is happening "under the hood" if I do not do that.

**Reply by lnicola** [reply]
Yeah, not sure what to say about that. Keep in mind that `coc-rust-analyzer` is a fork of the official VS Code extension, so unless you can reproduce the issue with Code or another client, this is probably not the best place to ask about issues.
